### PR TITLE
[CL][tests]: Add all remaining intended test vectors from original tracking issue

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #5428

## What is the purpose of the change

This PR adds the remaining test vectors from #5428. No new issues were encountered while implementing these, but the existing infinite loop bug was hit by two test vectors that are left commented out in this PR.

## Testing and Verifying

- Added test cases can be found in `position_test.go`
- A few minor fixes were made to testing logic in `range_test.go`

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/range_test.go [hunk 1]
```diff
@@ -312,29 +408,46 @@ func (s *KeeperTestSuite) trackEmittedIncentives(cumulativeTrackedIncentives sdk
...
 		if recordStartTime.Before(lastTrackerUpdateTime) {
 			// If the record started emitting prior to the last incentiveCreationTime (the last time we checkpointed),
 			// then we assume it has been emitting for the whole period since then.
-			secondsEmitted = int64(s.Ctx.BlockTime().Sub(lastTrackerUpdateTime).Seconds())
+			secondsEmitted = sdk.NewDec(int64(s.Ctx.BlockTime().Sub(lastTrackerUpdateTime))).QuoInt64(int64(time.Second))
 		} else if recordStartTime.Before(s.Ctx.BlockTime()) {
 			// If the record started emitting between the last incentiveCreationTime and now, then we only track the
 			// emissions between when it started and now.
-			secondsEmitted = int64(s.Ctx.BlockTime().Sub(recordStartTime).Seconds())
+			secondsEmitted = sdk.NewDec(int64(s.Ctx.BlockTime().Sub(recordStartTime))).QuoInt64(int64(time.Second))
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/range_test.go:419]: Note to reviewer: the original version was truncating the seconds to int, leading to incorrect expected amounts in some cases

**Review by czarcas7ic — APPROVED** [review]
Nice work 👍

File: x/concentrated-liquidity/range_test.go [hunk 2]
```diff
@@ -266,6 +349,19 @@ func (s *KeeperTestSuite) executeRandomizedSwap(pool types.ConcentratedPoolExten
...
+		pool, err := s.clk.GetPoolById(s.Ctx, pool.GetId())
+		s.Require().NoError(err)
+
+		poolSpotPrice := pool.GetCurrentSqrtPrice().Power(2)
+		minSwapOutAmount := poolSpotPrice.Mul(sdk.SmallestDec()).TruncateInt()
+		poolBalances := s.App.BankKeeper.GetAllBalances(s.Ctx, pool.GetAddress())
+		if poolBalances.AmountOf(swapOutDenom).LTE(minSwapOutAmount) {
+			return sdk.Coin{}, sdk.Coin{}
+		}
+	}
```
Comment (p0mvn) [inline x/concentrated-liquidity/range_test.go:363]: Why is this only done for swapOutDenom of token 1?

**Review by p0mvn — APPROVED** [review]
LGTM. Nice work

I haven't tried running each test case yet to make sure that it tests what it intends to test. However, the diff looks sound to me

File: x/concentrated-liquidity/range_test.go (hunk 2 above)
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/range_test.go:363]: This specific methodology (multiplying by sqrt price, which is asset1 / asset0) only applies for asset1, but I agree this check could be added in the other direction as well

**Comment by czarcas7ic** [comment]
@AlpinYukseloglu this broke main I believe, and we also did not back port this

**Pull request merged**